### PR TITLE
feat: added Wishlist Page

### PR DESCRIPTION
**Pull request by srv-pranjal**
# Added Wishlist Page

## Description

 - Added Wishlist Provider and useWishlist custom hook.
 - Added Add to wishlist, remove from wishlist and get user wishlist Utilities
 - Created a single Horizontal Product Card Component for Cart and Wishlist
 - Added Move to Cart and Move to Wishlist Functionalities
 - Handled the case while moving product if already present in Cart or wishlist
 
## Add to Wishlist
image

## Wishlist
image

## Move to Cart
image

### CODE REVIEW

**Review by tewarimohit — APPROVED** [review]
Good to merge.....

**Review by sharmayatendra — APPROVED** [review]
code looks clean
design is consistent
folder structure is good

**Review by pradipgharge — APPROVED** [review]
The code is clean, Naming is also very nice. Looks good to merge

File: src/pages/Cart/Cart.js [hunk 1]
```diff
@@ -20,17 +20,27 @@ export const Cart = () => {
...
+              View your Wishlist
+            </Link>
+            or
+            <Link to="/products" className="btn btn--link-text" role="button">
+              Continue Shopping
+            </Link>
+          </p>
         </section>
       )}
     </main>
```
Comment (vivek-guda) [inline src/pages/Cart/Cart.js:46]: I like the use of semantic HTML

**Comment by vivek-guda** [comment]
Things I like:
Code looks clean 
UI looks good
Use of semantic HTML
Folder structure is good

**Pull request merged**